### PR TITLE
Add a list_projects command to the API/CLI

### PR DESCRIPTION
**Pull request by ryabouza**
In response to issue #355, add a new API call to list all projects
and the nodes connected to them.

@henn, is this what you were looking for?

### CODE REVIEW

File: haas/api.py [hunk 1]
```diff
@@ -661,6 +661,22 @@ def port_detach_nic(port):
...
+    """List all projects and nodes that belong to them.
+
+    Returns a JSON array of objects representing a list of projects.
+
+    Example:  '[{"nodes": ["node1", "node2"], "name": "project1"},
+                {"nodes": ["node3", "node4"], "name": "project2"}]'
+    """
+    db = model.Session()
+    projects = db.query(model.Project).all()
+    projects = [{'name': p.label, 'nodes': [n.label for n in p.nodes]}
```
Comment (haydarkarrar) [inline haas/api.py]: We have another API call for getting the nodes of the project. I think returning only list of projects is sufficient.

File: haas/api.py [hunk 2]
```diff
@@ -661,6 +661,22 @@ def port_detach_nic(port):
...
 
 
+@rest_call('GET', '/projects')
+def list_projects():
+    """List all projects and nodes that belong to them.
+
+    Returns a JSON array of objects representing a list of projects.
+
+    Example:  '[{"nodes": ["node1", "node2"], "name": "project1"},
+                {"nodes": ["node3", "node4"], "name": "project2"}]'
```
Comment (haydarkarrar) [inline haas/api.py]: I don't think so. We can write another API call to get headnode of the project.
Example:
`/projects/<project_name>/headnode`

**Comment by zenhack** [comment]
Re: @haydarkarrar's suggestions, I don't really feel very strongly -- I think we should either do this "information dump" style and give a detailed view of the projects, or reduce it to just the names, but I don't really care which. My temptation is to leave the decision to whoever steps up and implements it, but I want to get others to weigh in first. @gsilvis, @lokI8, thoughts?

**Comment by zenhack** [comment]
Ah, just noticed, this is beeing worked on as part of @haydarkarrar's pr (#405).

**Comment by gsilvis** [comment]
The simplest thing to do is have 'list_projects' just give a list of projects (admin only, once we have auth) and 'show_project' show all the details of a project (can be run by someone with access to that project).  We need the latter call anyways.

**Comment by henn** [comment]
I like the idea of doing a "dump" of everything: nets, headnodes, nodes, etc. This will allow external systems (schedulers, GUIs, etc) to get the status of everything without needing to make a bunch of separate calls.

These calls would add overhead and increase latency.

**Comment by gsilvis** [comment]
Superseded

**Pull request closed without merging**